plugins/module_utils/ndb/vlans.py

```diff
@@ -0,0 +1,311 @@
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+from copy import deepcopy
+
+from .clusters import get_cluster_uuid
+from .nutanix_database import NutanixDatabase
+
+__metaclass__ = type
+
+
+class VLAN(NutanixDatabase):
+    def __init__(self, module, is_stretched=False):
+        resource_type = "/resources/networks"
+        super(VLAN, self).__init__(module, resource_type=resource_type)
+        self.is_stretched = is_stretched
+        self.build_spec_methods = {
+            "name": self._build_spec_name,
+            "desc": self._build_spec_desc,
+            "vlan_type": self._build_spec_type,
+            "ip_pools": self._build_spec_ip_pools,
+            "gateway": self._build_spec_gateway,
+            "subnet_mask": self._build_spec_subnet_mask,
+            "primary_dns": self._build_spec_primary_dns,
+            "secondary_dns": self._build_spec_secondary_dns,
+            "dns_domain": self._build_spec_dns_domain,
+            "cluster": self._build_spec_cluster,
+            "vlans": self._build_spec_vlans,
+        }
+
+    def get_spec(self, old_spec=None, params=None):
```

Comment:
Check update get_spec(). It has **kwargs which can be used to switch between builders for Vlans and strecthed VLans.

Reply:
Or we can have seperate class for Vlans and StretchedVlans(Vlans).